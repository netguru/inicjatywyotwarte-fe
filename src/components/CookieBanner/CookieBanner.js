import React from 'react';
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import SummaryButton from 'components/Buttons/SummaryButton/SummaryButton'

const Banner = styled.div`
  position: fixed;
  bottom: 10px;
  left: 0;
  right: 0;
  margin: 0 auto;
  max-width: 90%;
  padding: 25px;
  font-size: 0.9rem;
  background: var(--main-color-category-neighbour);
  color: var(--font-color-medium);
  z-index: 10;
  opacity: 0.97;
  text-align: justify;
  line-height: 1.4rem;
  border-radius: var(--border-radius-medium);
`

const StyledLink = styled(Link)`
  text-decoration: none;
  color: var(--blue-active-color);
`

const StyledSummaryButton = styled(SummaryButton)`
  margin-bottom: 0;
`

const getInitialCookieAccepted = () => {
  return localStorage.getItem('areCookiesAccepted') || false;
}

const CookieBanner = () => {
  const [cookieAccepted, setCookieAccepted] = React.useState(getInitialCookieAccepted)

  const acceptCookie =() => {
    localStorage.setItem('areCookiesAccepted', 'true')
    setCookieAccepted(true);
  }

  return (
    !cookieAccepted && (
    <Banner>
      <>
        <span>
          Netguru SA korzysta z plików cookies w celu analizowania ruchu na
          stronie. Przez dalsze korzystanie z naszego Serwisu, bez zmian
          ustawień w zakresie prywatności, wyrażasz dobrowolną zgodę na
          zapisywanie plików cookies w Twojej przeglądarce. Więcej
          informacji na temat plików cookies znajdziesz w naszej&nbsp;
          {
            <StyledLink to='/polityka-prywatnosci'>
              “Polityce Prywatności"
            </StyledLink>
          }
          .
        </span>
        <StyledSummaryButton onClick={acceptCookie}>
          Akceptuję
        </StyledSummaryButton>
      </>
    </Banner>
  )
)}

export default CookieBanner
