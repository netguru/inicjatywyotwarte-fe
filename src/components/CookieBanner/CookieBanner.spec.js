import React from 'react'
import CookieBanner from './CookieBanner'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('CookieBanner', () => {
  test('renders loader', () => {
    const wrapper = shallow(<CookieBanner/>);
    expect(wrapper.find('Banner').length).toEqual(1);
  });

  test('has proper text', () => {
    const wrapper = shallow(<CookieBanner/>);
    expect(wrapper.find('Banner').find('span').text()).toEqual(
      "Netguru SA korzysta z plików cookies w celu analizowania ruchu na stronie. Przez dalsze korzystanie z naszego Serwisu, bez zmian ustawień w zakresie prywatności, wyrażasz dobrowolną zgodę na zapisywanie plików cookies w Twojej przeglądarce. Więcej informacji na temat plików cookies znajdziesz w naszej “Polityce Prywatności\"."
    );
  });

  test('has proper lint', () => {
    const wrapper = shallow(<CookieBanner/>);
    expect(wrapper.find('Styled(Link)').props().to).toEqual('/polityka-prywatnosci')
  });


  test('match snapshot', () => {
    const wrapper = shallow(<CookieBanner/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should save to localStorage', () => {
    const wrapper = shallow(<CookieBanner/>);
    expect(localStorage.areCookiesAccepted).toEqual(undefined)
    wrapper.find('Styled(SummaryButton)').simulate('click');
    expect(localStorage.areCookiesAccepted).toEqual("true")
  });
});


