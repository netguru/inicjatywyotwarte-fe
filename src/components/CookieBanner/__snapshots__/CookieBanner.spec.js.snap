// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookieBanner match snapshot 1`] = `
<Banner>
  <span>
    Netguru SA korzysta z plików cookies w celu analizowania ruchu na stronie. Przez dalsze korzystanie z naszego Serwisu, bez zmian ustawień w zakresie prywatności, wyrażasz dobrowolną zgodę na zapisywanie plików cookies w Twojej przeglądarce. Więcej informacji na temat plików cookies znajdziesz w naszej 
    <Styled(Link)
      to="/polityka-prywatnosci"
    >
      “Polityce Prywatności"
    </Styled(Link)>
    .
  </span>
  <Styled(SummaryButton)
    onClick={[Function]}
  >
    Akceptuję
  </Styled(SummaryButton)>
</Banner>
`;
